mne/preprocessing/tests/test_ica.py

```diff
@@ -1079,6 +1079,34 @@ def test_fit_params(method, tmpdir):
         assert ica.fit_params == fit_params_after_instantiation
 
 
+@pytest.mark.parametrize(
+    ('param_name', 'param_val'),
+    (
+        ('start', 0),
+        ('stop', 500),
+        ('decim', 2),
+        ('reject', dict(eeg=500e-6)),
+        ('flat', dict(eeg=1e-6))
+    )
+)
+def test_fit_params_epochs_vs_raw(param_name, param_val):
+    """Check that we get a warning when passing parameters that get ignored."""
+    method = 'infomax'
+    n_components = 3
+    max_iter = 1
+
+    raw = read_raw_fif(raw_fname).pick_types(meg=False, eeg=True)
+    events = read_events(event_name)
+    epochs = Epochs(raw, events=events)
+    ica = ICA(n_components=n_components, max_iter=max_iter, method=method)
+
+    fit_params = {param_name: param_val}
+
+    ica.fit(inst=raw, **fit_params)  # should not warn
```

Comment:
might it be overkill to fit on `raw` 5 times just to show that no warning is raised?  Surely those kwargs are tested with raw in some other test already?

Reply:
I can kill the `raw` test here

Reply:
Resolved in 2b388ff